File: PESDK_LITE(OpenSource)_Android_220623_1.5.3.520278/PEUISDK/src/main/java/com/pesdk/uisdk/analyzer/internal/SegmentationEngine.java
package com.pesdk.uisdk.analyzer.internal;

import android.graphics.Bitmap;

/**
 *
 */
public interface SegmentationEngine {
    /**
     * 创建分割引擎
     *
     * @param mode      仅MattingEngine 时参数有效,  0 人像抠图; 1 天空抠图
     * @param binPath
     * @param paramPath
     */
    void createAnalyzer(int mode, String binPath, String paramPath);

    /**
     * 释放
     */
    void release();


    /**
     * 异步
     */
    void asyncProcess(Bitmap bitmap, OnSegmentationListener listener);

    /**
     * 同步
     */
    void syncProcess(Bitmap bitmap, OnSegmentationListener listener);

}

File: PESDK_LITE(OpenSource)_Android_220623_1.5.3.520278/VEEngine/src/main/java/com/vesdk/engine/segmentation/SegmentationEngine.kt
package com.vesdk.engine.segmentation

import android.graphics.Bitmap
import com.vecore.matting.MattingerFactory
import com.vesdk.engine.bean.base.BaseSegmentationConfig
import com.vesdk.engine.listener.OnEngineSegmentationListener

/**
 * 分割引擎
 */
abstract class SegmentationEngine {

    companion object {

        /**
         * 人像
         */
        const val MODEL_PORTRAIT = MattingerFactory.MattingerOption.PORTRAIT_MATTING

        /**
         * 天空
         */
        const val MODEL_SKY = MattingerFactory.MattingerOption.SKY_MATTING

        /**
         * 使用bitmap
         */
        const val MODE_USE_BITMAP_RESULT = 3

        /**
         * 使用原始数据
         */
        const val MODE_USE_RAW_BUFFER_RESULT = 4

    }

    /**
     * 配置相同
     */
    abstract fun isSameConfig(config: BaseSegmentationConfig): Boolean

    /**
     * 创建
     */
    abstract fun create()

    /**
     * 释放
     */
    abstract fun release()

    /**
     * 异步
     */
    abstract fun asyncProcess(bitmap: Bitmap, listener: OnEngineSegmentationListener)

    /**
     * 同步
     */
    abstract suspend fun syncProcess(bitmap: Bitmap): Bitmap

}